pkg/build/builder/dockerutil.go

```diff
@@ -156,15 +156,27 @@ func dockerRun(client DockerClient, createOpts docker.CreateContainerOptions, lo
 
 	containerName := containerNameOrID(c)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
```

Comment:
> It does not leak go routines.

IIUC it does leak a goroutine if there is no signal received.

Code from [pkg/util/interrupt/interrupt.go](https://github.com/kubernetes/kubernetes/blob/62eb7d3db8b0d968bd3346d067bb8f58e086352b/pkg/util/interrupt/interrupt.go), with my own inline comments:

```go
func (h *Handler) Run(fn func() error) error {
	ch := make(chan os.Signal, 1)
	signal.Notify(ch, childSignals...)
	defer signal.Stop(ch)
	go func() {
		sig, ok := <-ch  // will block forever if we go out of the scope of `Run` without having received a signal.
		if !ok {
			// unreachable code, no code ever closes ch.
			return
		}
		h.Signal(sig)
	}()
	defer h.Close()
	return fn()
}
```

The reason it can block forever is that we have no code closing `ch` to notify the goroutine to terminate.
After calling `signal.Stop(ch)`, we know that the runtime won't send anything through the `ch` channel, so the receive operation will block forever, leaking the goroutine ∎ (Q.E.D.)

I'll send a patch to Kube.